packages/ai/src/authorizers/federated-connections/index.ts

```diff
@@ -7,12 +7,14 @@ export type { AsyncStorageValue } from "./asyncLocalStorage";
 
 export { asyncLocalStorage };
 
-export type FederatedConnectionAuthorizerParams<ToolExecuteArgs extends any[]> =
-  {
-    refreshToken: AuthorizerToolParameter<ToolExecuteArgs, string>;
-    scopes: AuthorizerToolParameter<ToolExecuteArgs, string[]>;
-    connection: AuthorizerToolParameter<ToolExecuteArgs, string>;
-  };
+export type FederatedConnectionAuthorizerParams<ToolExecuteArgs extends any[]> = {
+  refreshToken?: AuthorizerToolParameter<ToolExecuteArgs, string>;
+  getAccessToken?: (
```

Comment:
why not use `AuthorizerToolParameter<ToolExecuteArgs, string>;` for this as well, this can also be a function or value if its setup in a some non singleton scenario